lucene/analysis/common/src/java/org/apache/lucene/analysis/util/AnalysisSPILoader.java

```diff
@@ -143,6 +144,24 @@ public S newInstance(String name, Map<String,String> args) {
   public Set<String> availableServices() {
     return originalNames;
   }  
+
+  /**
+   * Looks up SPI name (static "NAME" field) with appropriate modifiers.
+   * Also it must be a String class and declared in the concrete class.
+   * @return the SPI name
+   * @throws NoSuchFieldException - if the "NAME" field is not defined.
+   * @throws IllegalAccessException - if the "NAME" field is inaccessible.
+   * @throws IllegalStateException - if the "NAME" field does not have appropriate modifiers or isn't a String field.
+   */
+  public static String lookupSPIName(Class<? extends AbstractAnalysisFactory> service) throws NoSuchFieldException, IllegalAccessException, IllegalStateException {
```

Comment:
This could be package private (since the original method was not public and still this is called only from classes in the same package)?

Reply:
I thought that this method is useful if you want to figure out the name of a factory where you only have class reference. There is no reason to hide it.